src/execution_plan/optimizations/utilize_indices.c

```diff
@@ -195,6 +209,55 @@ bool _simple_predicates(const FT_FilterNode *filter) {
 	return true;
 }
 
+static bool _validateInExpression(AR_ExpNode *exp) {
+	if(!strcasecmp(exp->op.func_name, "in") == 0) return false;
+	assert(exp->op.child_count == 2);
+	AR_ExpNode *list = exp->op.children[1];
+	if(list->type != AR_EXP_OPERAND || list->operand.type != AR_EXP_CONSTANT) return false;
+	assert(list->operand.constant.type == T_ARRAY);
+	SIValue listValue = list->operand.constant;
+	if(SIArray_Length(listValue) == 0) return false;
+	uint listLen = SIArray_Length(listValue);
+	for(uint i = 0; i < listLen; i++) {
+		// Clone value since the array might be free later.
+		SIValue v = SI_CloneValue(SIArray_Get(listValue, i));
+		// Ignore everything other than number and strings.
+		if(!(SI_TYPE(v) & (SI_NUMERIC | T_STRING | T_BOOL))) return false;
+	}
+	return true;
+}
+
+static void _transformInToOrSequence(OpFilter *filter) {
+	AR_ExpNode *inOp = filter->filterTree->exp.exp;
+	AR_ExpNode *lhs = rm_malloc(sizeof(AR_ExpNode));
+	memcpy(lhs, inOp->op.children[0], sizeof(AR_ExpNode));
+	AR_ExpNode *list = inOp->op.children[1];
+	SIValue listValue = list->operand.constant;
+	uint listLen = SIArray_Length(listValue);
+	AR_ExpNode *constant = AR_EXP_NewConstOperandNode(SIArray_Get(listValue, 0));
+	FT_FilterNode *root = FilterTree_CreatePredicateFilter(OP_EQUAL, lhs, constant);
+	for(uint i = 1; i < listLen; i ++) {
+		FT_FilterNode *orNode = FilterTree_CreateConditionFilter(OP_OR);
+		AppendLeftChild(orNode, root);
+		constant = AR_EXP_NewConstOperandNode(SIArray_Get(listValue, i));
+		lhs = rm_malloc(sizeof(AR_ExpNode));
+		memcpy(lhs, inOp->op.children[0], sizeof(AR_ExpNode));
+		AppendRightChild(orNode, FilterTree_CreatePredicateFilter(OP_EQUAL, lhs, constant));
+		root = orNode;
+	}
+	FilterTree_Free(filter->filterTree);
+	filter->filterTree = root;
+}
+
+static void _prepareFilterOp(OpFilter *filter) {
+// Filter is applicable, normalize it.
+	_normalize_filter(filter->filterTree);
+	// See if the filter tree needed to be modified, if so, replace the original, since the op will be free.
+	if(filter->filterTree->t == FT_N_EXP &&
+	   strcasecmp(filter->filterTree->exp.exp->op.func_name, "in") == 0)
```

Comment:
Create a local function which performs this check.

Reply:
done